Libplanet/PoS/Model/Delegation.cs

```diff
@@ -46,5 +57,32 @@ public IValue Serialize()
                 .Add(DelegatorAddress.Serialize())
                 .Add(ValidatorAddress.Serialize());
         }
+
+        public override bool Equals(object? obj)
+        {
+            return Equals(obj as Delegation);
+        }
+
+        public bool Equals(Delegation? other)
+        {
+            return !(other is null) &&
+                   Address.Equals(other.Address) &&
+                   DelegatorAddress.Equals(other.DelegatorAddress) &&
+                   ValidatorAddress.Equals(other.ValidatorAddress);
+        }
+
+        public override int GetHashCode()
+        {
+            int code = 0;
+            unchecked
+            {
+                foreach (byte b in Address.ToByteArray())
+                {
+                    code = (code * 397) ^ b.GetHashCode();
+                }
+            }
+
+            return code;
+        }
```

Comment:
Could we use `ByteUtil`? 
```suggestion
        public override int GetHashCode() => ByteUtil.CalculateHashCode(ToByteArray());
```
and implement `ToByteArray()`

Reply:
Seems much better 👍

Reply:
I've overrided some other `GetHashCode()` of classes that using validator address as a key, with other prime number (not 397) to avoid hash collision between them.
Actually, I think it's not so much matters if hash collision occurs between different type of classes, but not sure since I don't know much about its use cases.
If hash codes used for just collections, I think it would be better to use same prime number(397) on other classes, and use `ByteUtil.CalculateHashCode()`.
Do you have any opinion for this?